internal/nfs/nodeserver/nodeserver.go

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeserver
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	csicommon "github.com/ceph/ceph-csi/internal/csi-common"
+	"github.com/ceph/ceph-csi/internal/util"
+	"github.com/ceph/ceph-csi/internal/util/log"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	mount "k8s.io/mount-utils"
+	netutil "k8s.io/utils/net"
+)
+
+const (
+	defaultMountPermission = uint64(0o777)
```

Comment:
777 is wider permission not possible to get rid of it? we removed 777 for both cephfs and rbd

Reply:
I'll use `0o750` here as default too ?
https://github.com/ceph/ceph-csi/blob/8c5563a9bcef229b6cfe54ae9fbfbbbed70089be/internal/util/util.go#L313-L316

Reply:
```
[1mSTEP[0m: create a PVC and bind it to an app with normal user
Jul 28 11:20:37.978: INFO: Waiting up to &PersistentVolumeClaim{ObjectMeta:{nfs-2540  nfs-2540    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:PersistentVolumeClaimSpec{AccessModes:[ReadWriteMany],Resources:ResourceRequirements{Limits:ResourceList{},Requests:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},},VolumeName:,Selector:nil,StorageClassName:*csi-nfs-sc,VolumeMode:nil,DataSource:nil,DataSourceRef:nil,},Status:PersistentVolumeClaimStatus{Phase:,AccessModes:[],Capacity:ResourceList{},Conditions:[]PersistentVolumeClaimCondition{},AllocatedResources:ResourceList{},ResizeStatus:nil,},} to be in Bound state
Jul 28 11:20:37.978: INFO: waiting for PVC nfs-2540 (0 seconds elapsed)
Jul 28 11:20:39.983: INFO: waiting for PVC nfs-2540 (2 seconds elapsed)
Jul 28 11:20:39.990: INFO: Waiting for PV pvc-aab980e4-1de5-4e70-882b-acaa1e8ed549 to bind to PVC nfs-2540
Jul 28 11:20:39.990: INFO: Waiting up to timeout=10m0s for PersistentVolumeClaims [nfs-2540] to have phase Bound
Jul 28 11:20:39.992: INFO: PersistentVolumeClaim nfs-2540 found and phase=Bound (2.726822ms)
Jul 28 11:20:39.992: INFO: Waiting up to 10m0s for PersistentVolume pvc-aab980e4-1de5-4e70-882b-acaa1e8ed549 to have phase Bound
Jul 28 11:20:39.995: INFO: PersistentVolume pvc-aab980e4-1de5-4e70-882b-acaa1e8ed549 found and phase=Bound (2.674095ms)
Jul 28 11:20:40.009: INFO: Waiting up to pod-run-as-non-root to be in Running state
Jul 28 11:20:56.020: INFO: ExecWithOptions {Command:[/bin/sh -c echo testing > /target/testing] Namespace:nfs-2540 PodName:pod-run-as-non-root ContainerName:write-pod Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Jul 28 11:20:56.020: INFO: >>> kubeConfig: /root/.kube/config
Jul 28 11:20:56.021: INFO: ExecWithOptions: Clientset creation
Jul 28 11:20:56.021: INFO: ExecWithOptions: execute(POST https://192.168.39.123:8443/api/v1/namespaces/nfs-2540/pods/pod-run-as-non-root/exec?command=%2Fbin%2Fsh&command=-c&command=echo+testing+%3E+%2Ftarget%2Ftesting&container=write-pod&container=write-pod&stderr=true&stdout=true)
Jul 28 11:20:56.187: INFO: stdErr occurred: /bin/sh: /target/testing: Permission denied

Jul 28 11:20:56.187: FAIL: failed to validate normal user CephFS pvc and application binding: failed to exec command in pod: command terminated with exit code 1
```

Hey @Madhu-1 , I dont think that will work. 
https://github.com/kubernetes-csi/csi-driver-nfs/blob/c78a98288f3d648cf9d6147fa310444a7f228640/cmd/nfsplugin/main.go#L31
I'll revert to what the csi-nfs-driver used previously 0777.

Reply:
Do we have securityContext set in the application pod yaml?

Reply:
:+1: